tests/tensorflow2/test_keras_mirrored.py

```diff
@@ -390,12 +390,13 @@ def test_training_end(out_dir, tf_eager_mode):
 
 
 @pytest.mark.slow
-def test_include_regex(out_dir, tf_eager_mode):
+@pytest.mark.parametrize("workers", ["one", "all"])
+def test_include_regex(out_dir, tf_eager_mode, workers):
     hook = KerasHook(
         out_dir=out_dir,
         save_config=SaveConfig(save_interval=9),
         include_collections=["custom_coll"],
-        include_workers="all",
+        include_workers=workers,
```

Comment:
do the number of tensors remain 4 when workers=one and workers=all? probably should vary with strategy.num_replicas_in_sync in the case of workers=all

Reply:
can you check if we have an issue in general with save_all_workers in TF eager mode? maybe parameterizing all the tests to test workers="one" and workers="all" would help to check quickly

Reply:
> do the number of tensors remain 4 when workers=one and workers=all? probably should vary with strategy.num_replicas_in_sync in the case of workers=all

The number of tensors will remain the same.
I have added an assert to count the number of workers for each tensor

Reply:
> can you check if we have an issue in general with save_all_workers in TF eager mode? maybe parameterizing all the tests to test workers="one" and workers="all" would help to check quickly

Are we worried at all about slowing down the test suite by doing this though?

Reply:
maybe add this parameterization only for test_save_all?

Reply:
There seems to be three tests that seems to be doing the same thing:

`test_save_all_workers`, `test_save_one_worker` and `test_save_all_workers`.

Maybe we should coalesce these three by just parameterizing `test_save_all_workers`?

Reply:
sounds good